### PR TITLE
Enable consistency for ghcr.io registry

### PR DESCRIPTION
**Pull request by ChristianBieri1995**
## What problem does this PR solve?

The following PR enables consistency over all chaos-mesh components for fetching images from the ghcr.io registry.

## What's changed and how it works?

All components from chaos-mesh which use the ghcr.io repository to fetch images extract the predifined variable from helpers.tpl (for example chaos-dashboard.image, line 145-148 in helpers.tpl) - except the dns component. 

In order to reach consistency, the following changes are needed:
<ol>
  <li>add chaos-dns.image variable to helpers.tpl</li>
  <li>change structure of .Values.dnsServer.image in order to construct chaos-dns.image in helpers.tpl correctly</li>
  <li>change dns-deployment.yaml in order to use variable from helpers.tpl</li>
</ol>

## Related changes

- [ ] This change also requires further updates to the [website](https://github.com/chaos-mesh/website) (e.g. docs)
- [ ] This change also requires further updates to the `UI interface`

## Cherry-pick to release branches (optional)

> This PR should be cherry-picked to the following release branches:

- [x] release-2.6
- [ ] release-2.5

### CHANGELOG

> Must include at least one of them.

- [ ] I have updated the `CHANGELOG.md`
- [x] I have labeled this PR with "no-need-update-changelog"

### Tests

> Must include at least one of them.

- [ ] Unit test
- [ ] E2E test
- [x] Manual test

### Side effects

- [ ] **Breaking backward compatibility**

## DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

File: helm/chaos-mesh/values.yaml [hunk 1]
```diff
@@ -405,8 +405,14 @@ dnsServer:
...
-  # Image of DNS Server
-  image: ghcr.io/chaos-mesh/chaos-coredns:v0.2.6
+  # image would be constructed by <registry>/<repository>:<tag>
+  image:
+    # override global registry, empty value means using the global images.registry
+    registry: ""
+    # repository part for image of chaos-dashboard
+    repository: chaos-mesh/chaos-coredns
+    # override global tag, empty value means using the global images.tag
+    tag: ""
```
Comment (STRRL) [inline helm/chaos-mesh/values.yaml:415]: it seems we should use the tag "v0.2.6" here.

chaos-coredns is a standalone component that does not get a new version released when Chaos Mesh core components release.

Comment (STRRL) [reply helm/chaos-mesh/values.yaml:415]: rest LGTM!

and I do not prefer cherrypick it into the release-2.6, it contains a breaking changes in helm chart, it changes the schema of the field `dnsServer.image` from string to object

Comment (ChristianBieri1995) [reply helm/chaos-mesh/values.yaml:415]: Great, thank you for checking my PR. Your advise also LGTM!

Comment (g1eny0ung) [reply helm/chaos-mesh/values.yaml:415]: Hi @ChristianBieri1995, do you have time to update this one place? And, you also have to fix the DCO check and update the changelog.

Comment (ChristianBieri1995) [reply helm/chaos-mesh/values.yaml]: Updated, I will fix the DCO check later as soon as I have access to my local machine.

**Review by g1eny0ung — APPROVED** [review]
LGTM!

**Review by cwen0 — APPROVED** [review]

**Pull request merged**